openapi2conv/openapi2_conv.go

```diff
@@ -638,6 +641,49 @@ func FromV3Swagger(swagger *openapi3.Swagger) (*openapi2.Swagger, error) {
 	return result, nil
 }
 
+func consumesToArray(consumes map[string]bool) []string {
+	var consumesArr []string
+	for key, _ := range consumes {
+		consumesArr = append(consumesArr, key)
+	}
+	sort.Strings(consumesArr)
+	return consumesArr
+}
+
+func FromV3RequestBodies(name string, requestBodyRef *openapi3.RequestBodyRef, components *openapi3.Components) (*openapi2.Parameters, *openapi2.Parameters, map[string]bool, error) {
+	formParameters := openapi2.Parameters{}
+	bodyOrRefParameters := openapi2.Parameters{}
+	if ref := requestBodyRef.Ref; ref != "" {
+		bodyOrRefParameters = append(bodyOrRefParameters, &openapi2.Parameter{Ref: FromV3Ref(ref)})
+		return &bodyOrRefParameters, &formParameters, nil, nil
+	}
+
+	//Only select one formData or request body for an individual requesstBody as swagger 2 does not support multiples
+	consumes := make(map[string]bool)
```

Comment:
```suggestion
	var consumes map[string]struct{}
```
Let's not allocate if requestBodyRef.Value == nil.

Reply:
I'm not sure I follow, it never allocates if  ref.Value == nil.  It would have returned an empty map. I use a map[string]bool just for the ease of accessing the key in the KVP

Reply:
Doesn't `make(...)` perform allocation? It is called before the `if`, regardless.